crates/yas_core/src/tests/utils/swap_cases.cairo

```diff
@@ -0,0 +1,505 @@
+        MIN_TICK, MAX_TICK, MIN_SQRT_RATIO, MAX_SQRT_RATIO
+    };
+    use yas_core::numbers::fixed_point::implementations::impl_64x96::{
+        FP64x96Impl, FixedType, FixedTrait
+    };
+    use yas_core::numbers::signed_integer::{
+        i32::i32, i32::i32_div_no_round, i256::i256, integer_trait::IntegerTrait
+    };
+    use yas_core::tests::utils::constants::FactoryConstants::{tick_spacing, FeeAmount, fee_amount};
+
+    use integer::BoundedInt;
+
+    #[derive(Copy, Drop, Serde)]
+    struct SwapTestCase {
+        zero_for_one: bool,
+        has_exact_out: bool,
+        exact_out: bool,
+        amount_specified: i256,
+        sqrt_price_limit: FixedType,
+    }
+
+    #[derive(Drop, Serde)]
+    struct PoolTestCase {
+        fee_amount: u32,
+        starting_price: FixedType,
+        mint_positions: Array<Position>
+    }
+
+    #[derive(Copy, Drop, Serde)]
+    struct Position {
+        tick_lower: i32,
+        tick_upper: i32,
+        liquidity: u128
+    }
+
+    #[derive(Copy, Drop, Serde)]
+    struct SwapExpectedResults {
+        amount_0_before: u256,
+        amount_0_delta: i256,
+        amount_1_before: u256,
+        amount_1_delta: i256,
+        execution_price: u256,
+        fee_growth_global_0_X128_delta: u256,
+        fee_growth_global_1_X128_delta: u256,
+        pool_price_after: u256,
+        pool_price_before: u256,
+        tick_after: i32,
+        tick_before: i32,
+    }
+
+    #[derive(Copy, Drop, Serde)]
+    struct SwapExpectedResultsError {
+        pool_balance_0: u256,
+        pool_balance_1: u256,
+        pool_price_before: FixedType,
+        tick_before: i32,
+    }
+
+    fn obtain_swap_cases(idxs: Array<u32>) -> (Array<SwapTestCase>, Array<SwapTestCase>) {
+        let mut success_cases = array![];
+        let mut error_cases = array![];
+
+        let mut i = 0;
+        loop {
+            if i == SWAP_CASES().len() {
+                break;
+            }
+            if i < idxs.len() {
+                if contains(@idxs, i) {
+                    success_cases.append(*SWAP_CASES()[i]);
+                } else {
+                    error_cases.append(*SWAP_CASES()[i]);
+                }
+            } else {
+                error_cases.append(*SWAP_CASES()[i]);
+            }
+            i += 1;
+        };
+        (success_cases, error_cases)
+    }
+
+    //this function checks if number is in array
+    fn contains(arr: @Array<u32>, value: u32) -> bool {
+        let mut i = 0;
+        let mut result = false;
+        loop {
+            if i == arr.len() {
+                break;
+            }
+            if *arr[i] == value {
+                result = true;
+                break;
+            }
+            i += 1;
+        };
+        result
+    }
+
+    fn POOL_CASES() -> Array<PoolTestCase> {
+        array![
+            PoolTestCase {
+                // description: 'low fee, 1:1 price, 2e18 max range liquidity',
```

Comment:
```suggestion
```

Reply:
all these are the pool case description, are we sure we want to delete them?